dev/archery/archery/docker/core.py

```diff
@@ -56,6 +50,18 @@ def _sanitize_command(cmd):
     'arm64v8': 'aarch64',
 }
 
+_arrow_root = os.environ.get(
+    'ARROW_ROOT',
+    os.path.abspath(__file__).rsplit("/", 5)[0]
```

Comment:
The default value will only work if archery was installed with the `-e` option and on windows `__file__` will use `\`.
It might still be a sensible default as I think installing with `-e` should be the default for devs and we also use it in CI.

```suggestion
    os.path.abspath(__file__).rsplit(os.path.sep, 5)[0]
```

Reply:
Ah. That pattern is actually copied from https://github.com/apache/arrow/blob/master/dev/archery/archery/integration/util.py#L39-L42
Perhaps we should factor it out...

Reply:
Ok, done.